services/motion/builtin/move_request.go

```diff
@@ -837,3 +838,25 @@ func toGeoPosePlanSteps(posesByComponent []motionplan.PlanStep, geoPoses []spati
 	}
 	return steps, nil
 }
+
+// This function will check the orientation of the startPose of a 2d move request, and ensure that it is normal to the XY plane. If it is
+// not, it will be altered such that it is (accounting for e.g. an ourdoor rover with one wheel on a rock). If the orientation is such that
+// the robot is pointed directly up or down (or is upside-down), an error is returned.
+func correctStartPose(startPose spatialmath.Pose) (spatialmath.Pose, error) {
+	orient := startPose.Orientation().OrientationVectorRadians() // OV is easy to check if we are in plane
+	if orient.OZ < 0 {
+		return nil, errors.New("rover appears to be upside down, please correct")
+	}
+	if orient.OX == 0 && orient.OY == 0 {
+		return startPose, nil
+	}
+
+	adjPt := spatialmath.NewPoseFromPoint(r3.Vector{0, 1, 0})
+	newAdjPt := spatialmath.Compose(spatialmath.NewPoseFromOrientation(orient), adjPt).Point()
+	if 1-math.Abs(newAdjPt.Z) < 0.0001 {
+		return nil, errors.New("rover appears pointing either straight up or down, check your movement sensor")
```

Comment:
```suggestion
		return nil, errors.New("base appears pointing either straight up or down, check your movement sensor")
```

Reply:
These are going to be user-facing errors, so mind please being specific about whether it appears the base is straight up or down?

Reply:
Done